src/rules/at-function-parentheses-space-before/index.js

```diff
@@ -0,0 +1,42 @@
+import { utils } from "stylelint";
+import { namespace } from "../../utils";
+
+export const ruleName = namespace("at-function-parentheses-space-before");
+
+export const messages = utils.ruleMessages(ruleName, {
+  rejected: "Unexpected whitespace before parentheses in function declaration",
+  expected: "Expected a single space before parentheses in function declaration"
+});
+
+export default function(value) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: value,
+      possible: ["always", "never"]
+    });
+    if (!validOptions) {
+      return;
+    }
+
+    root.walkAtRules(decl => {
```

Comment:
I think that you should be able to do the `function` filtering already here, e.g.
https://github.com/kristerkari/stylelint-scss/blob/34668406e422148029bada5d32ef6405d11b7c5f/src/rules/at-extend-no-missing-placeholder/index.js#L18